src/Sarif/Core/SarifLog.cs

```diff
@@ -51,18 +51,24 @@ public static SarifLog Load(string sarifFilePath)
         /// <returns>SarifLog instance for file</returns>
         public static SarifLog Load(Stream source, bool deferred = false)
         {
-            JsonSerializer serializer = new JsonSerializer();
+            var serializer = new JsonSerializer();
 
             if (deferred)
             {
                 serializer.ContractResolver = new SarifDeferredContractResolver();
+                var jptr = new JsonPositionedTextReader(Stream.Synchronized(source));
+                return serializer.Deserialize<SarifLog>(jptr);
             }
-
-            using (StreamReader sr = new StreamReader(source))
-            using (JsonTextReader jtr = new JsonTextReader(sr))
+            else
             {
```

Comment:
you return in the previous conditional clause, so why use the else here? you can get rid of this clause and left shift the code (in terms of indentation).

Reply:
done